### ISSUE TITLE
strange issue with world generated blocks

### ISSUE THREAD

**Issue by Smoker120900** [FirEmerald/DynamicVerticalSlabsAndStairs#103]
![image](https://github.com/user-attachments/assets/a4704731-78a7-4cda-8f2f-9d6d5e4bc032)
hello, is this a known issue?? (only happens with stair blocks)

**Comment by FirEmerald** [comment]
I don't even know HOW that happens...

**Comment by Smoker120900** [comment]
no idea, i just entered a world and all structures that have stair blocks are bugged like this, the only way to fix that i found (in-game) is by breaking one of the stairs that's connected to the other broken stairs, that fixes a line of aligned stairs, but not all, only the ones that are "connected" to the one i break

**Comment by Smoker120900** [comment]
btw, another thing i noticed is this: (ignore the fact i'm playing in spanish)

https://github.com/user-attachments/assets/6dba3ca4-9774-4906-a2b4-74b81181451e

is the lighting difference noticeable?? it happens when using the center or the border of the placement grid (i'm using embeddium and embeddium dynamic lights, if this helps more to understand the issue)

**Comment by Smoker120900** [comment]
forgot to mension:
i'm using MC 1.20.1, Forge (latest), Mod's latest version

**Comment by Morenthus** [comment]
> is the lighting difference noticeable?? it happens when using the center or the border of the placement grid (i'm using embeddium and embeddium dynamic lights, if this helps more to understand the issue)

Ran into the same thing, happens on any version ≥2.0.0 and rolling back to 1.8.0 fixed it

**Comment by Jusey1** [comment]
Ye'h, having same issue with world generated stair blocks.

**Comment by ffuentesm** [comment]
This is also happening on all stairs on generated structures on neoforge 1.21 too 
![image](https://github.com/user-attachments/assets/4c9127fc-93c2-4f18-9fb4-ac06fcac5592)

**Comment by BlockerOBP** [comment]
Here is another sample for this issue with forge 1.20.1, the latest version 2.1.0
![2024-11-26_11 43 39](https://github.com/user-attachments/assets/a203f5ef-d184-40ee-ba06-5cfa3a4d021b)

**Comment by BlockerOBP** [comment]
I can confirm that in version 2.0.8 on forge 1.20.1 the problem is no longer present
![2024-11-26_12 04 40](https://github.com/user-attachments/assets/2d8ca0fe-7e35-4e42-8a2d-759b75ea547f)

**Comment by SmallPpBigWorld** [comment]
> I can confirm that in version 2.0.8 on forge 1.20.1 the problem is no longer present ![2024-11-26_12 04 40](https://private-user-images.githubusercontent.com/189693044/389788707-2d8ca0fe-7e35-4e42-8a2d-759b75ea547f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI2ODI5MDMsIm5iZiI6MTczMjY4MjYwMywicGF0aCI6Ii8xODk2OTMwNDQvMzg5Nzg4NzA3LTJkOGNhMGZlLTdlMzUtNGU0Mi04YTJkLTc1OWI3NWVhNTQ3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QwNDQzMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OGUyMTQwZGQ0OWNhMWUzNDNkMjAzOWE3MGNiMjgxMGU4ODhhYTc2YTY2ZGQ3MTExM2E0ZWU4ZGRjMDMyZjNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZRFfSURBfpaKchvF4m9oy1NmivNVdBml7INZPx0chNU)

also a non-issue in 1.21.1 fabric with 2.0.8 while 2.1.0 does present with that issue.

**Comment by nanakisan** [comment]
I was wondering if it was just me. Glad to see it wasn't the case. I made an observation that having the additional placements mode enabled this happens. But if you explore while it is turned off then stairs work as they should when generating.

**Comment by TimmyWithTheBat21** [comment]
also happens for me i am on neoforge 1.21.1. what strange is that not all structures seams to be affected some are some dont. well for the time beeing i am gone switch to the previous version ...

**Comment by Smoker120900** [comment]
ok, so the lighting issue is gone (no idea how, maybe i updated a mod and it fixed the issue), but the world generated stairs are still broken, i tried disabling the additional placement mode as default for every time i enter a world to see if that worked, but no

**Comment by ffuentesm** [comment]
@Smoker120900 are you using "Snow Real Magic" in your mod list when this error happens?

**Comment by Smoker120900** [comment]
> @Smoker120900 are you using "Snow Real Magic" in your mod list when this error happens?

nah, btw the issue got fixed, maybe it was one of the optimization mods i have
